CSharp/README.md

```diff
@@ -23,6 +24,115 @@ var httpClient = new HttpClient();
 IUserService userService = new UserService();
 ```
 
+### Don't use explicit *this* reference
+
+It is redundant and does not add for readability. For more information read [this SO post](http://stackoverflow.com/questions/5885249/to-use-or-not-to-use-the-this-qualifier-in-c-sharp).
+
+```csharp
+// Bad
+this.ValidateParameters();
+
+// Good
+ValidateParameters();
+```
+
+### Use *string* instead of *String* keyword
+
+```csharp
+// Bad
+String.IsNullOrEmpty(name);
+
+// Good
+string.IsNullOrEmpty(name);
+```
+
+### Use readonly fields when possible
+
+### Use string interpolation
+
+```csharp
+// Bad
+var message = string.Format("My name is {0} {1}.", person.FirstName, person.LastName);
+
+// Good
+var message = $"My name is {person.FirstName} {person.LastName}.");
+```
+
+## Layout
+
+### Use *four spaces* per indentation level
+
+```csharp
+// Bad
+public void SomeMethod()
+{
+  DoSomethingFirst();
+  DoSomerthingLater();
+}
+
+// Good
+public void SomeMethod()
+{
+    DoSomethingFirst();
+    DoSomerthingLater();
+}
+```
+
+### Remove unused or redudant namespaces
+
+### Use single line and lambda getters
+
+```csharp
+// Good
+public string Name { get; private set; }
+
+// Good
+public string UppercaseName => Name.toUpperCase(); 
+```
+
+### Don't use braces for just one line

```

Comment:
Do you have some links about that? While I understand the point, it kind of gets in the way of early return patterns where you can simplify a lot the readability of some code by performing simple checks and returning early, something that we also want to encourage.

This is what we have in mind:

```csharp
if (!Model.Valid()) throw new InvalidSomethingException();
if (Model.DoNothing) return;
if (DBModel.WasAlreadyUpdated) return;

// perform update and changes...
```

Maybe, for cases like this, we could encourage the statement to be written in the same line as the condition.